### PR TITLE
V7.3.12

### PR DESCRIPTION
**Pull request by mattip**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Update the recipe. I removed `version` from the jinja2 template since python3.8 does not have a 7.3.12 release. Looking at the [changelog](https://doc.pypy.org/en/latest/release-v7.3.12.html#changelog) for 7.3.12, I don't see anything that is critical for 3.8, but the backported fixes for functools may allow some progress with the pandas feedstock. Anyhow I think CPython and PyPy python3.8 will be dropped soon after the migration of CPython 3.12 starts, which should not too far in the future.

I would like to add a solution for #101 as well.

### CODE REVIEW

**Comment by mattip** [comment]
CI is green. Anyone want to review?

File: recipe/bld.bat [hunk 1]
```diff
@@ -108,10 +85,17 @@ REM timeout 60m pypy3 -m test --pgo -j%CPU_COUNT% || true;
...
+    pypy -c "import _ctypes_test"
+    IF %ERRORLEVEL% NEQ 0 (Echo ERROR while building &exit /b 11)
+    pypy -c "import _testmultiphase"
+) else (
+    pypy -c "import _ctypes_test_build"
+    IF %ERRORLEVEL% NEQ 0 (Echo ERROR while building &exit /b 11)
+    pypy -c "import _testmultiphase_build"
+)
 IF %ERRORLEVEL% NEQ 0 (Echo ERROR while building &exit /b 11)
+copy
```
Comment (h-vetinari) [inline recipe/bld.bat:98]: That looks like a leftover?

File: recipe/bld.bat [hunk 2]
```diff
@@ -126,3 +110,9 @@ pypy -m venv destination
 destination\Scripts\python -c "import sqlite3"
 if errorlevel 1 exit 1
 rmdir /q /s destination
+
+REM Copy required DLLs so pypy can be used without activating the environment
+copy /b %PREFIX%\Library\bin\libexpat.dll %PREFIX%
+copy /b %PREFIX%\Library\bin\libbz2.dll %PREFIX%
+copy /b %PREFIX%\Library\bin\ffi-8.dll %PREFIX%
+
```
Comment (h-vetinari) [inline recipe/bld.bat:118]: Extra newline

File: recipe/build.sh [hunk 3]
```diff
@@ -137,10 +121,19 @@ echo INCLUDEPY $(pypy -c "from distutils import sysconfig; print(sysconfig.get_c
 ls $(pypy -c "from distutils import sysconfig; print(sysconfig.get_config_var('INCLUDEPY'))")
 
 _PYTHON_SYSCONFIGDATA_NAME=$sysconfigdata_name pypy -c "from distutils import sysconfig; assert sysconfig.get_config_var('HOST_GNU_TYPE') != None"
-# Build the c-extension modules for the standard library
+# Build the test c-extension modules
+# PyPy uses a wrapper for _ctypes_test to trick the import system into gettig
+# the c-extension from `/tmp`, so we need to copy it into LIB
```
Comment (h-vetinari) [inline recipe/build.sh:126]:
```suggestion
# PyPy uses a wrapper for _ctypes_test to trick the import system into getting
# the c-extension from `/tmp`, so we need to copy it into LIB 
```

**Review by h-vetinari — APPROVED** [review]
This LGTM (aside from small nits; did not verify the patches), but as I haven't followed the discussion in #101, I'll leave the approval to @isuruf.

File: recipe/bld.bat [hunk 4]
```diff
@@ -126,3 +109,8 @@ pypy -m venv destination
 destination\Scripts\python -c "import sqlite3"
 if errorlevel 1 exit 1
 rmdir /q /s destination
+
+REM Copy required DLLs so pypy can be used without activating the environment
+copy /b %PREFIX%\Library\bin\libexpat.dll %PREFIX%
+copy /b %PREFIX%\Library\bin\libbz2.dll %PREFIX%
+copy /b %PREFIX%\Library\bin\ffi-8.dll %PREFIX%
```
Comment (mattip) [inline recipe/bld.bat:116]: @isuruf this is the part that copies the 3 dlls in repsonse to #101

Comment (isuruf) [reply recipe/bld.bat:116]: We shouldn't copy them in this package. The copies will have precedence over the ones in `%PREFIX%\Library\bin` which get upgraded through their own conda packages.

Comment (isuruf) [reply recipe/bld.bat:116]: Otherwise, looks good to me.

**Comment by mattip** [comment]
I commented out the fix for #101. That issue will have to wait for another day.

**Comment by mattip** [comment]
@conda-forge-admin, please rerender

**Comment by mattip** [comment]
Since the controversial fix for #101 was removed, I will merge this.

**Comment by h-vetinari** [comment]
Quick question: are you planning to do the aarch/ppc builds like for the last version bump (#99)?

**Comment by mattip** [comment]
I want to make some progress with [pandas](https://github.com/pandas-dev/pandas/issues/42509#issuecomment-1674726205) first, perhaps there are more problems in the feedstock that need to be fixed.

**Pull request merged**